datafusion/datasource-parquet/src/source.rs

```diff
@@ -438,6 +438,22 @@ impl ParquetSource {
     }
 }
 
+/// Parses datafusion.common.config.ParquetOptions.coerce_int96 String to a arrow_schema.datatype.TimeUnit
+fn parse_coerce_int96_string(str_setting: &str) -> datafusion_common::Result<TimeUnit> {
+    let str_setting_lower: &str = &str_setting.to_lowercase();
+
+    match str_setting_lower {
+        "ns" => Ok(TimeUnit::Nanosecond),
+        "us" => Ok(TimeUnit::Microsecond),
+        "ms" => Ok(TimeUnit::Millisecond),
+        "s" => Ok(TimeUnit::Second),
```

Comment:
nit: Do we really need the `millis` and `secs` resolutions?

Reply:
I don't know of any systems that do this, but I didn't take my absence of knowledge as proof some system out there doesn't do it. This basically just matches the behavior that we put into arrow-rs.

Reply:
fair enough